SHMEM/params.h

```diff
@@ -43,7 +44,11 @@ POSSIBILITY OF SUCH DAMAGE.
 
 // The data type used for the keys
 // If you change this, you will have to change the SHMEM API calls used
+#ifdef UINT32_KEYS
+typedef uint32_t KEY_TYPE;
+#else
 typedef int KEY_TYPE;
```

Comment:
`int` and `uint32_t` are the same size.  Do you want to support different size keys?  Also, I would suggest that all key types have the same signed-ness (signed or unsigned).